src/services/synthetix.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+/* eslint-disable i18next/no-literal-string */
+
+import { utils, Contract } from 'ethers';
+import { BigNumber } from 'bignumber.js';
+
+import { getEnv } from 'configs/envConfig';
+
+import { getContract, encodeContractMethod } from 'services/assets';
+
+// utils
+import { getEthereumProvider, parseTokenBigNumberAmount, reportOrWarn } from 'utils/common';
+import { isProdEnv } from 'utils/environment';
+import { parseOffer, createAllowanceTx } from 'utils/exchange';
+
+import { PROVIDER_SYNTHETIX } from 'constants/exchangeConstants';
+
+// models
+import type { Asset } from 'models/Asset';
+import type { Offer } from 'models/Offer';
+import type { AllowanceTransaction } from 'models/Transaction';
+
+// assets
+import ERC20_CONTRACT_ABI from 'abi/erc20.json';
+import RATES_ABI from 'abi/synthetixRates.json';
+import EXCHANGE_ABI from 'abi/synthetixExchange.json';
+
+const ethProvider = getEthereumProvider(getEnv().NETWORK_PROVIDER);
+
+const exchangeAddress = '0xC011a73ee8576Fb46F5E1c5751cA3B9Fe0af2a6F';
+const ratesAddress = isProdEnv
+  ? '0xda80E6024bC82C9fe9e4e6760a9769CF0D231E80'
+  : '0xA36f5A656c48EB0b43b63293E690DA746162d40B';
+
+export const createSynthetixOrder = async (
+  fromAsset: Asset, toAsset: Asset, amount: string | number, clientSendAddress: string,
+): Promise<Object> => {
+  try {
+    const data = encodeContractMethod(
+      EXCHANGE_ABI,
+      'exchange',
+      [
+        utils.formatBytes32String(fromAsset.symbol),
+        parseTokenBigNumberAmount(amount, fromAsset.decimals),
+        utils.formatBytes32String(toAsset.symbol),
+      ],
+    );
+
+    if (!data) throw new Error();
```

Comment:
Could we an an error message within the `new Error` please? I appreciate that we could trace the call stack but for tools like Sentry, it's more beneficial to have a message here.